taichi/codegen/llvm/llvm_codegen_utils.cpp

```diff
@@ -50,10 +49,54 @@ bool is_same_type(llvm::Type *required, llvm::Type *provided) {
     }
     return true;
   }
-  auto req_name = type_name(required);
-  auto prov_name = type_name(provided);
-  int min_len = std::min(req_name.size(), prov_name.size());
-  return req_name.substr(0, min_len) == prov_name.substr(0, min_len);
+
+  auto a_name = type_name(a);
```

Comment:
nit: it'd be nice to add a brief note here (or better place if applicable) explaining how the names are produced and formatted. From the comments below I can tell `xxx.111, xxx, xxx.222` are same types but it'd be nice to have a note so that it's easier to understand ;)